semantic-tags/src/Tags/Tagging/Precise.hs

```diff
@@ -38,6 +58,62 @@ runTagging source
   . execWriter
   . runReader source
 
--- | Slices a range out of 'Source' and gives back the first line of source up to 180 characters.
-firstLine :: Source -> Range -> Text
-firstLine src = Text.stripEnd . Text.take 180 . Text.takeWhile (/= '\n') . Source.toText . slice src
+-- | Takes a Loc (where the span's column offset is measured in bytes) and
+-- returns two Spans: A 1-indexed span LSP friendly span (where column offset is
+-- measure in utf16 code units).
+calculateLineAndSpans ::
+  Source -> -- |^ Source
+  Loc ->    -- |^ Location of identifier
+  (Text, OneIndexedSpan, UTF16CodeUnitSpan)
+calculateLineAndSpans src Loc {byteRange = srcRange, span = span@Span {start = start@Pos {column = startCol}, end = end@Pos {column = endCol}}} =
```

Comment:
Nitpick: can we add some whitespace to this record pattern match up a little bit? There’s a lot of information there and vertical spacing would be good.